### PR TITLE
feat: add support for HttpRequest[[]byte]

### LINKED ISSUE #849: GET requests seem to be parsing query params incorrectly
This module:

```go
type EchoRequest struct {
	Name string `json:"name"`
}

type EchoResponse struct {
	Message string `json:"message"`
}

//ftl:verb
//ftl:ingress http GET /idv/echo
func Echo(ctx context.Context, req builtin.HttpRequest[EchoRequest]) (builtin.HttpResponse[EchoResponse], error) {
	return builtin.HttpResponse[EchoResponse]{
		Status: 200,
		Body:   EchoResponse{Message: fmt.Sprintf("Hello, %s!", req.Body.Name)},
	}, nil
}
```

returns the error:
```bash
curl -i http://localhost:8892/ingress/idv/echo\?name\=wes
HTTP/1.1 400 Bad Request
Content-Type: text/plain; charset=utf-8
Vary: Origin
X-Content-Type-Options: nosniff
Date: Tue, 30 Jan 2024 20:46:56 GMT
Content-Length: 94

builtin.HttpRequest<idv.EchoRequest>.body.name has wrong type, expected String found []string
```

### LINKED ISSUE #850: HTTP ingress not using new `alias` annotations
```go
type EchoResponse struct {
	Message string `alias:"wes"`
}

//ftl:verb
//ftl:ingress http GET /idv/echo
func Echo(ctx context.Context, req builtin.HttpRequest[EchoRequest]) (builtin.HttpResponse[EchoResponse], error) {
	return builtin.HttpResponse[EchoResponse]{
		Status: 200,
		Body:   EchoResponse{Message: "Hello!"},
	}, nil
}
```

Responds with:

```bash
curl -i http://localhost:8892/ingress/idv/echo
HTTP/1.1 200 OK
Vary: Origin
Date: Tue, 30 Jan 2024 20:55:22 GMT
Content-Length: 20
Content-Type: text/plain; charset=utf-8

{"Message":"Hello!"}% 
```

### LINKED ISSUE #856: Support `Bytes` as request and response payload.

### PR DESCRIPTION
**Pull request by wesbillman**
Fixes #849
Fixes #850 
Fixes #856

### CODE REVIEW

File: backend/controller/ingress/ingress.go [hunk 1]
```diff
@@ -168,7 +169,46 @@ func ValidateAndExtractRequestBody(route *dal.IngressRoute, r *http.Request, sch
 	return body, nil
 }
 
-func buildRequest(route *dal.IngressRoute, r *http.Request, dataRef *schema.DataRef, sch *schema.Schema) (map[string]any, error) {
+func extractHTTPRequestBody(route *dal.IngressRoute, r *http.Request, dataRef *schema.DataRef, sch *schema.Schema) (any, error) {
+	data, err := sch.ResolveDataRefMonomorphised(dataRef)
```
Comment (alecthomas) [inline backend/controller/ingress/ingress.go:173]: Nice.

File: backend/schema/schema.go [hunk 2]
```diff
@@ -57,6 +57,23 @@ func (s *Schema) ResolveDataRef(ref *DataRef) *Data {
...
 
+func (s *Schema) ResolveDataRefMonomorphised(ref *DataRef) (*Data, error) {
+	data := s.ResolveDataRef(ref)
+	if data == nil {
+		return nil, fmt.Errorf("unknown data %v", ref)
+	}
+
+	if len(ref.TypeParameters) > 0 {
+		var err error
+		data, err = data.Monomorphise(ref.TypeParameters...)
```
Comment (alecthomas) [inline backend/schema/schema.go:68]: MBD but you could just do `return data.Monomorphise(ref.TypeParameters...)` here

File: examples/go/httpingress/httpingress.go [hunk 3]
```diff
@@ -19,15 +19,17 @@ type GetResponse struct {
...
 func Get(ctx context.Context, req builtin.HttpRequest[GetRequest]) (builtin.HttpResponse[GetResponse], error) {
 	logger := ftl.LoggerFromContext(ctx)
 	logger.Infof("Path: %s", req.Path)
 	logger.Infof("Method: %s", req.Method)
-	logger.Infof("Query: %s", req.Query)
-	logger.Infof("Body: %s", req.Body)
-	logger.Infof("Headers: %s", req.Headers)
+	logger.Infof("Query: %v", req.Query)
+	logger.Infof("Body: %v", req.Body)
+	logger.Infof("Headers: %v", req.Headers)
```
Comment (alecthomas) [inline examples/go/httpingress/httpingress.go:32]: Maybe this should be debug?

**Review by alecthomas — APPROVED** [review]

**Comment by alecthomas** [comment]
Nice! I took a bit of a crack at this yesterday, but this is much cleaner than what I came up with!

File: examples/go/httpingress/httpingress.go (hunk 3 above)
Comment (wesbillman) [reply examples/go/httpingress/httpingress.go:32]: Yeah I was mostly just using them to make testing this httpIngress stuff super easy and show the logger, but none of this is actually needed :)

**Review by alecthomas — APPROVED** [review]
Awesome! Thanks Wes.

**Pull request merged**